packages/engine/Source/Renderer/Texture.js

```diff
@@ -448,6 +505,35 @@ function loadImageSource(texture, source) {
   );
 }
 
+/**
+ * Load texel data from an Image into part of a texture
+ *
+ * @param {Texture} texture The texture to which texel values will be loaded.
+ * @param {ImageData|HTMLImageElement|HTMLCanvasElement|HTMLVideoElement} source The source for texel values to be loaded into the texture.
+ * @param {number} xOffset The texel x coordinate of the lower left corner of the subregion of the texture to be updated.
+ * @param {number} yOffset The texel y coordinate of the lower left corner of the subregion of the texture to be updated.
+ *
+ * @private
+ */
+function loadPartialImageSource(texture, source, xOffset, yOffset) {
+  const context = texture._context;
+  const gl = context._gl;
+
+  gl.pixelStorei(gl.UNPACK_ALIGNMENT, 4);
+  gl.pixelStorei(gl.UNPACK_PREMULTIPLY_ALPHA_WEBGL, texture.preMultiplyAlpha);
+  gl.pixelStorei(gl.UNPACK_FLIP_Y_WEBGL, texture.flipY);
+
+  gl.texSubImage2D(
```

Comment:
Maybe I'm just confused by the documentation, but [are height and with not required for a WebGL 2 context](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/texSubImage2D)?

Reply:
The WebGL1 signature is [still supported in WebGL2](https://registry.khronos.org/webgl/specs/latest/2.0/#3.7.6:~:text=interface%20mixin%20WebGL2RenderingContextOverloads). Basically WebGL2 is [almost 100% backwards compatible with WebGL1](https://webgl2fundamentals.org/webgl/lessons/webgl1-to-webgl2.html), so any function that does not explicitly use WebGL2 features (aside from some extensions) will still work.